### DISCUSSION TITLE
Relax SS conditions for osu!mania

### DISCUSSION THREAD

**Discussion by mcendu** [ppy/osu#28299] [Ideas] [answered]
Currently getting an SS requires hitting all Perfects (MAX or flashing 300), which is extremely difficult given the frame perfect timing required.

Stable also awards SS to scores that also has Great (yellow 300) hits, and I suggest lazer should follow suit.

Example code:

```cs
        public override ScoreRank RankFromScore(double accuracy, IReadOnlyDictionary<HitResult, int> results)
        {
            if (results.Where(result => result.Key < HitResult.Great).All(result => result.Value == 0))
                return ScoreRank.X;

            return base.RankFromScore(accuracy, results);
        }
```

**Comment by bdach** [comment, marked as answer]
please read https://github.com/ppy/osu/issues/24238
